ksql-engine/src/main/java/io/confluent/ksql/schema/inference/DefaultSchemaInjector.java

```diff
@@ -130,7 +134,9 @@ private static CreateSource addSchemaFields(
       final ConfiguredStatement<CreateSource> preparedStatement
   ) {
     try {
-      return TableElement.fromSchema(schema);
+      schema.fields()
+          .forEach(field -> LogicalSchemas.toSqlTypeConverter().validate(field.schema()));
```

Comment:
it needs to be field by field to be backwards compatible (and to avoid failing unit tests). The `ConnectSchemaTranslator` has the following code:
```java
  private static Schema toKsqlStructSchema(final Schema schema) {
    final SchemaBuilder schemaBuilder = SchemaBuilder.struct();
    for (final Field field : schema.fields()) {
      try {
        final Schema fieldSchema = toKsqlFieldSchema(field.schema());
        schemaBuilder.field(field.name().toUpperCase(), fieldSchema);
      } catch (final UnsupportedTypeException e) {
        log.error("Error inferring schema at field {}: {}", field.name(), e.getMessage());
      }
    }
    return schemaBuilder.optional().build();
  }
```
This means that if there's ever only unsupported types (e.g. BYTES), then we will return a struct with no fields. This fails validation at the top level, but not at each field (there are no fields). This might be a different bug, but that's out of scope.

Reply:
Oh also test coverage is in `DefaultSchemaInjectorFunctionalTest`

Reply:
I'm not sure I follow. We should jump on a call.